action-binding-generator/src/main/kotlin/io/github/typesafegithub/workflows/actionbindinggenerator/typing/TypingGeneration.kt

```diff
@@ -76,8 +76,12 @@ private fun EnumTyping.buildEnumCustomType(
     fieldName: String,
     className: String,
 ): TypeSpec {
-    val itemsNames = itemsNames ?: items.map { it.toPascalCase() }
-    val itemsNameMap = items.zip(itemsNames).toMap()
+    val itemsNames =
+        itemsNames ?: items
+            .map { it.toPascalCase() }
+            .groupBy { it }
+            .values
+            .flatMap { it.mapIndexed { index, name -> "$name${if (index > 0) (index + 1) else ""}" } }
```

Comment:
How about going with:
```suggestion
            .flatMap { it.mapIndexed { index, name -> "$name_${if (index > 0) (index + 1) else ""}" } }
```
to visually separate the extra part? Especially if the input/output ends with a number (which is rare, but it does in our tests 😄).

Reply:
Besides that your change does not compile, I get the point.
I indeed thought about the same and decided against, but it was a close decision without strong opinion, so I changed it to contain the underscore now. :-)
If a value ends in digits like in the test this is most prominent actually.